math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -2977,6 +3004,45 @@ RVec<T> Construct(const RVec<Args_t> &... args)
    return ret;
 }
 
+/// Enumerate method:
+/// For any Rvec v produce another RVec with entries starting from 0, and incrementing by 1 until a N = v.size() is reached.
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// RVecF v = {1., 2., 3.};
+/// cout << Enumerate(v1) << "\n";
+/// // { 0, 1, 2 }
+template <typename T>
+RVec<T> Enumerate(const RVec<T> &v)
+{
+   const auto size = v.size();
+   RVec<T> ret;
+   ret.reserve(size);
+   for (auto i = 0UL; i < size; ++i) {
+      ret.emplace_back(i);
+   }
+   return ret;
+}
+
+/// Range method: 
+/// Produce RVec with entries starting from 0, and incrementing by 1 until a user-specified N is reached.
+/// Example code, at the ROOT prompt:
+/// ~~~{.cpp}
+/// using namespace ROOT::VecOps;
+/// cout << Range(3) << "\n";
+/// // { 0, 1, 2 }
+template <typename T>
+RVec<T> Range(const T &length)
```

Comment:
this should not return a RVec<T> but a RVec<RVec<T>::size_type>, and `length` can just be a `std::size_t` (taken by value, not by const ref)